Audiobook Player/Player/SleepTimer.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  SleepTimerManager.swift
+//  Audiobook Player
+//
+//  Created by Florian Pichler on 30.03.18.
+//  Copyright © 2018 Florian Pichler.
+//
+
+import UIKit
+import Foundation
+
+enum SleepTimerState {
+    case ready
+    case running
+    case expired
+}
+
+typealias SleepTimerCompletion = () -> Void
+typealias SleepTimerStart = () -> Void
+typealias SleepTimerProgress = (Double) -> Void
+
+final class SleepTimer {
+    static let shared = SleepTimer()
+
+    let durationFormatter = DateComponentsFormatter()
+
+    var timer: Timer!
```

Comment:
what do you think about having this var as either:
`lazy var timer = Timer()`
or
`var timer: Timer?`
it would remove the `nil` checks and make the code look more 'swifty'(?). Let me know what you think

Reply:
Still learning swift, sounds good to me.